File: src/day1GoLang/two_crystal_balls.go
package main

import (
	"math"
)

func TwoCrystalBalls(arr [10000]bool) int {
	jumpAmount := int(math.Sqrt(float64(len(arr)))) 
	var i int 

	for i = jumpAmount; i < len(arr); i += jumpAmount {
		if arr[i] == true {
			break
		}
	}

	for j := i - jumpAmount; j < len(arr); j++ {
		if arr[j] == true {
			return j
		}
	}

	return -1 
}

File: src/day1/TwoCrystalBalls.ts
export default function two_crystal_balls(breaks: boolean[]): number {
	const jmpAmount = Math.floor(Math.sqrt(breaks.length));

	let i = jmpAmount;

	for (; i < breaks.length; i += jmpAmount) {
		if (breaks[i]) {
			break;
		}
	}

	if (i > breaks.length) {
		return -1;
	}

	i -= jmpAmount;

	for (; i < breaks.length; i++) {
		if (breaks[i]) {
			break;
		}
	}

	return i;
}
